flytekit/sensor/base_sensor.py

```
@@ -60,12 +62,14 @@ def __init__(
         for k, _ in signature.parameters.items():  # type: ignore
             annotation = type_hints.get(k, None)
             inputs[k] = annotation
+        metadata = TaskMetadata(timeout=timeout)
```

Should we raise a reasonable error in the case where the task metadata is set together with the timeout?

```python
FileSensor(timeout=12, metadata=TaskMetadata(...))
```

---

make sense, will add it

---

> Should we raise a reasonable error in the case where the task metadata is set together with the timeout?
> 
> ```python
> FileSensor(timeout=12, metadata=TaskMetadata(...))
> ```

Thank you!!!

---

Can you add a quick test with `pytest.raises` to check the error?

---

no problem, will add it now